youtube_dl/extractor/nhk.py

```diff
@@ -3,14 +3,98 @@
 import re
 
 from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+)
 
 
-class NhkVodIE(InfoExtractor):
+class NhkBaseInfoExtractor(InfoExtractor):
+    _API_URL_TEMPLATE = 'https://api.nhk.or.jp/nhkworld/%sod%slist/v7a/%s/%s/%s/all%s.json'
+
+    def _get_clean_field(self, episode, key):
+        return episode.get(key + '_clean') or episode.get(key)
+
+    def _list_episodes(self, lang, is_video, is_episode, m_id):
```

Comment:
it would be better to order the function parameter based on their importance(i think the most important one is `m_id`).